src/prefect/client/orchestration.py

```diff
@@ -3503,6 +3505,31 @@ async def update_flow_run_labels(
         )
         response.raise_for_status()
 
+    async def create_sla(self, sla: SlaTypes) -> UUID:
+        """
+        Creates a service level agreement.
+
+        Args:
+            sla: The SLA to create. Must have a deployment ID set.
+
+        Raises:
+            httpx.RequestError: if the SLA was not created for any reason
+
+        Returns:
+            the ID of the SLA in the backend
+        """
+        response = await self._client.post(
+            "/slas/",
+            json=sla.model_dump(mode="json", exclude_unset=True),
+        )
+        response.raise_for_status()
```

Comment:
Not 100% sure how we do this for triggers but do we want to maybe not raise if we can't create an SLA?

Since we're not making deployment 'success' dependent on creating the SLAs similar to triggers maybe it makes sense to report back that we couldn't create specific SLAs. That way we're not leaving the end user in a state of confusion about what SLAs got made. 

Long story short I think we should offer the end user the ability to explicitly know what SLAs were and weren't created and not have to determine that by the order that they put them in the list and some guess work.

Reply:
I see what you're saying here. If we wanted to trap this exception to throw one at the end of the SLA list, I think we'd probably handle that logic in `RunnerDeployment._create_slas`.

FWIW triggers raise one at a time if they fail to create, it's just not as explicit as I'm doing here (check out the `.create_automation` method on the client).